dbt_copilot_helper/commands/dns.py

```diff
@@ -247,140 +237,202 @@ def add_records(client, records, subdom_id, action):
     return response["ChangeInfo"]["Status"]
 
 
-def check_for_records(client, parent_id, subdom, subdom_id):
-    records_to_add = []
-    response = client.list_resource_record_sets(
-        HostedZoneId=parent_id,
-    )
+def copy_records_from_parent_to_subdomain(client, parent_zone_id, subdomain, zone_id):
+    response = client.list_resource_record_sets(HostedZoneId=parent_zone_id)
 
     for records in response["ResourceRecordSets"]:
-        if subdom in records["Name"]:
+        if subdomain in records["Name"]:
             click.secho(records["Name"] + " found", fg="green")
-            records_to_add.append(records)
-            add_records(client, records, subdom_id, "UPSERT")
-    return True
+            add_records(client, records, zone_id, "UPSERT")
 
 
-def create_hosted_zone(client, domain, start_domain, base_len):
-    parts = domain.split(".")
+def create_hosted_zones(client, base_domain, subdomain):
+    domains_to_create = get_required_subdomains(base_domain, subdomain)
+    zone_ids = {}
 
-    # We only want to create domains max 2 deep so remove all sub domains in excess
-    parts_to_remove = len(parts) - base_len - MAX_DOMAIN_DEPTH
-    domain_to_create = parts[parts_to_remove:]
-
-    # Walk back domain name
-    for x in reversed(range(len(domain_to_create) - (len(start_domain.split(".")) - 1))):
-        subdom = ".".join(domain_to_create[(x):]) + "."
+    for domain_name in domains_to_create:
+        parent_domain = domain_name.split(".", maxsplit=1)[1]
+        parent_zone = f"{parent_domain}."
+        parent_zone_id = _get_zone_id_or_abort(client, parent_zone)
 
         if not click.confirm(
             click.style("About to create domain: ", fg="cyan")
-            + click.style(f"{subdom}\n", fg="white", bold=True)
+            + click.style(f"{domain_name}\n", fg="white", bold=True)
             + click.style("Do you want to continue?", fg="cyan"),
         ):
             exit()
 
-        parent = ".".join(subdom.split(".")[1:])
-        response = client.list_hosted_zones_by_name()
-
-        for hz in response["HostedZones"]:
-            if hz["Name"] == parent:
-                parent_id = hz["Id"]
-                break
+        click.secho(f"Creating hosted zone for {domain_name}...", fg="yellow")
 
-        click.secho(f"Creating hosted zone for {subdom}...", fg="yellow")
         response = client.create_hosted_zone(
-            Name=subdom,
+            Name=domain_name,
             # Timestamp is on the end because CallerReference must be unique for every call
-            CallerReference=f"{subdom}_from_code_{int(time.time())}",
+            CallerReference=f"{domain_name}_from_code_{int(time.time())}",
         )
         ns_records = response["DelegationSet"]
-        subdom_id = response["HostedZone"]["Id"]
+        dom_zone_id = response["HostedZone"]["Id"]
```

Comment:
Could maybe rename `dom_zone_id`s to `subdomain_zone_id` for consistency with other names.